### PR TITLE
fix: close all proxy listeners

### LINKED ISSUE #23588: `TestDatabaseResource ` flakiness (timeouts)
#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/4514436827/jobs/7950522343

#### Relevant snippet

```
panic: test timed out after 10m0s
running tests:
	TestDatabaseResource (9m29s)
```

```
goroutine 22579 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc003e26ce8?)
	/opt/go/src/runtime/sema.go:62 +0x27
sync.(*WaitGroup).Wait(0xc003e26ce0)
	/opt/go/src/sync/waitgroup.go:116 +0xa5
github.com/gravitational/teleport/lib/service.(*LocalSupervisor).Wait(0xc001d22a80)
	/__w/teleport/teleport/lib/service/supervisor.go:322 +0x85
github.com/gravitational/teleport/tool/tctl/common.makeAndRunTestAuthServer.func1()
	/__w/teleport/teleport/tool/tctl/common/helpers_test.go:239 +0xa5
testing.(*common).Cleanup.func1()
	/opt/go/src/testing/testing.go:1150 +0x194
testing.(*common).runCleanup(0xc002e7e9c0, 0x0)
	/opt/go/src/testing/testing.go:1328 +0x1ea
testing.tRunner.func2()
	/opt/go/src/testing/testing.go:1570 +0x53
testing.tRunner(0xc002e7e9c0, 0x92155e0)
	/opt/go/src/testing/testing.go:1582 +0x255
created by testing.(*T).Run
	/opt/go/src/testing/testing.go:1629 +0x806
```

```
goroutine 23485 [select, 9 minutes]:
github.com/gravitational/teleport/lib/multiplexer.(*Listener).Accept(0xc0026c4060)
	/__w/teleport/teleport/lib/multiplexer/wrappers.go:135 +0x113
github.com/gravitational/teleport/lib/srv/alpnproxy.(*ListenerMuxWrapper).startAcceptingConnectionServiceListener(0xc002e04a00)
	/__w/teleport/teleport/lib/srv/alpnproxy/listener.go:98 +0xa2
created by github.com/gravitational/teleport/lib/srv/alpnproxy.NewMuxListenerWrapper
	/__w/teleport/teleport/lib/srv/alpnproxy/listener.go:56 +0x27b
```

### PR DESCRIPTION
**Pull request by nklaassen**
I have been seeing test timeouts where the alpn listener is blocked on an Accept. It seems the listener is not being closed.

I have used the following command to reproduce the timeouts, it passes with this change:

```
$ go test ./tool/tctl/common/ -run TestDatabaseResource --count 100
```

Fixes https://github.com/gravitational/teleport/issues/23588

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

**Review by mdwn — APPROVED** [review]

File: lib/service/service.go [hunk 1]
```diff
@@ -3067,15 +3040,18 @@ func (l *proxyListeners) Close() {
 	if !l.db.Empty() {
```
Comment (zmb3) [inline lib/service/service.go:3040]: is `Empty()` used anywhere else? The close method below is safe to call without the empty check, so I wonder if we can just remove it.

**Review by zmb3 — APPROVED** [review]

**Comment by nklaassen** [comment]
Noooooo this PR also hits a timeout in an alpnproxy Accept https://github.com/gravitational/teleport/actions/runs/4515080126/jobs/7951942541?pr=23591

File: lib/service/service.go (hunk 1 above)
Comment (nklaassen) [reply lib/service/service.go]: It is used in one other spot but yeah the call here is pointless, removed

**Comment by nklaassen** [comment]
this still seems the fix the TestDatabaseResource timeouts I was seeing, so hopefully it's a net improvement

**Pull request merged**